components/searchBarDestination/index.js

```diff
@@ -0,0 +1,132 @@
+import { SearchBar } from 'react-native-elements';
+import decodePolyline from 'decode-google-map-polyline';
+import styles from './styles';
+
+export default class searchBarDestination extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showPredictions: true,
+      destination: '',
+      predictions: [],
+      destinationRegion: {
+        latitude: 45.492409,
+        longitude: -73.582153,
+      },
+    };
+  }
+
+  async onChangeDestination(destination) {
+    this.setState({ destination });
+    const key = 'AIzaSyCqNODizSqMIWbKbO8Iq3VWdBcK846n_3w';
+    const apiUrl = `https://maps.googleapis.com/maps/api/place/autocomplete/json?key=${key}&input=${destination}&location=45.492409, -73.582153&radius=2000`;
+    try {
+      const result = await fetch(apiUrl);
+      const json = await result.json();
+      this.setState({
+        predictions: json.predictions
+      });
+    } catch (err) {
+      console.error(err);
+    }
+  }
+
+  async drawPath(prediction) {
+    // eslint-disable-next-line react/no-unused-state
+    this.setState({ description: prediction });
+    const key = 'AIzaSyCqNODizSqMIWbKbO8Iq3VWdBcK846n_3w';
+    const geoUrl = `https://maps.googleapis.com/maps/api/place/details/json?key=${key}&placeid=${prediction}`;
+
+    try {
+      const georesult = await fetch(geoUrl);
+      const gjson = await georesult.json();
+      this.setState({
+        destinationRegion: {
+          latitude: gjson.result.geometry.location.lat,
+          longitude: gjson.result.geometry.location.lng,
+
+        }
+      });
+      // eslint-disable-next-line no-shadow
+      const key = 'AIzaSyBsMjuj6q76Vcna8G5z9PDyTH2z16fNPDk';
+      // change to current location later
+      const originLat = this.props.updatedRegion.latitude === 0 ? 45.4656851 : this.props.updatedRegion.latitude;
+      const originLong = this.props.updatedRegion.longitude === 0 ? -73.7454814 : this.props.updatedRegion.longitude;
+      const destinationLat = this.state.destinationRegion.latitude;
+      const destinationLong = this.state.destinationRegion.longitude;
+      const directionUrl = `https://maps.googleapis.com/maps/api/directions/json?key=${key}&origin=${originLat},${originLong}&destination=${destinationLat},${destinationLong}`;
+      const result = await fetch(directionUrl);
+      const json = await result.json();
+      const encryptedPath = json.routes[0].overview_polyline.points;
+      const rawPolylinePoints = decodePolyline(encryptedPath);
+
+      // Incompatible field names for direct decode. Need to do a trivial conversion.
+      const waypoints = rawPolylinePoints.map((point) => {
+        return {
+          latitude: point.lat,
+          longitude: point.lng
+        };
+      });
+      this.props.coordinateCallback(waypoints);
+    } catch (err) {
+      console.error(err);
+    }
+  }
+
+  render() {
+    const placeholder = this.state.isMounted ? i18n.t('search') : 'Where do you want to go to?';
```

Comment:
Perhaps change to "Where to?" or "Where would you like to go?"

Reply:
@trevornag  the issues will be fixed shortly

Reply:
The second issue has been resolved, the application now sets the current location as default and the first search bar indicates that the current location is set.
We weren't able to fix the first problem for some reason it only works on android (we tried clearing the path when pressing the back button; however the functionality is the same)

Reply:
The issue has been resolved on my android, I can't debug apparently it doesn't work on Iphone